### PR TITLE
Fixed Iss599, GetFeatureInfo now works on leaflet and openlayers

### PR DESCRIPTION
**Pull request by MV88**
#599 In order to create a valid feature info request we create a bbox of 101x101 pixel that wrap the point.
The center point is repojected and a bbox of 101x101pixel is built around it.

Also added some functions in mapUtils and the tests don't throw errors.

### CODE REVIEW

File: web/client/utils/MapInfoUtils.js [hunk 1]
```diff
@@ -90,20 +77,86 @@ const MapInfoUtils = {
...
                     id: layer.id,
                     layers: layer.name,
                     query_layers: layer.name,
                     styles: layer.style,
-                    x: parseInt(props.point.pixel.x, 10),
-                    y: parseInt(props.point.pixel.y, 10),
-                    height: parseInt(props.map.size.height, 10),
-                    width: parseInt(props.map.size.width, 10),
-                    srs: CoordinatesUtils.normalizeSRS(crs),
+                    x: 51,
```
Comment (mbarto) [inline web/client/utils/MapInfoUtils.js]: let's use a constant, and make it overridable by props

File: web/client/utils/MapInfoUtils.js [hunk 2]
```diff
@@ -90,20 +77,86 @@ const MapInfoUtils = {
...
+    /**
+     * Create a new extent or update the provided extent.
+     * @param {number} minX Minimum X.
+     * @param {number} minY Minimum Y.
+     * @param {number} maxX Maximum X.
+     * @param {number} maxY Maximum Y.
+     * @param {Object} optExtent Destination extent.
+     * @return {Object Extent.
+     */
+    createOrUpdate(minX, minY, maxX, maxY, optExtent) {
```
Comment (mbarto) [inline web/client/utils/MapInfoUtils.js]: createOrUpdate what? It's better to have a meaningful name than a comment

File: web/client/utils/MapInfoUtils.js [hunk 3]
```diff
@@ -90,20 +77,86 @@ const MapInfoUtils = {
...
+        if (optExtent) {
```
Comment (mbarto) [inline web/client/utils/MapInfoUtils.js]: return assign(optExtent || {}, {minx: minX, ... })

File: web/client/utils/MapInfoUtils.js [hunk 4]
```diff
@@ -90,20 +77,86 @@ const MapInfoUtils = {
...
+     * Creates a bbox of size dimensions areund the center point given to it given the
+     * resolution and the rotation
+     * @param center {object} the x,y coordinate of the point
+     * @param resolution {number} the resolution of the map
+     * @param rotation {number} the optional rotation of the new bbox
+     * @param size {object} width,height of the desired bbox
+     * @param opt_extent {object}  optional bbox if passed it will be updated
+     * @return {object} the desired bbox {minx, miny, maxx, maxy}
+     */
+     getForViewAndSize(center, resolution, rotation = 0, size, optExtent) {
```
Comment (mbarto) [inline web/client/utils/MapInfoUtils.js]: getForViewAndSize what?

File: web/client/utils/MapInfoUtils.js [hunk 5]
```diff
@@ -90,20 +77,86 @@ const MapInfoUtils = {
...
+            optExtent);
+        return bbox;
+    },
     buildIdentifyRequest(layer, props) {
-        const {bounds, crs} = reprojectBbox(props.map.bbox, props.map.projection);
+        /* In order to create a valid feature info request
+         * we create a bbox of 101x101 pixel that wrap the point.
+         * center point is repojected then is built a box of 101x101pixel around it
+         */
+        let wrongLng = props.point.latlng.lng;
```
Comment (mbarto) [inline web/client/utils/MapInfoUtils.js]: Is this always wrong? I don't think so

File: web/client/utils/MapInfoUtils.js [hunk 6]
```diff
@@ -90,20 +77,86 @@ const MapInfoUtils = {
...
+        // longitude restricted to the -180,180 range
+        let lngCorrected = wrongLng - (360) * Math.floor(wrongLng / (360) + 0.5);
+        const center = {x: lngCorrected, y: props.point.latlng.lat};
+        let centerProjected = CoordinatesUtils.reproject(center, "EPSG:4326", "EPSG:900913");
+        const resolution = MapUtils.getCurrentResolution(props.map.zoom, 0, 21, 96);
+        const rotation = 0;
+        const size = [101, 101];
```
Comment (mbarto) [inline web/client/utils/MapInfoUtils.js]: Where is this used?

File: web/client/utils/MapInfoUtils.js (hunk 5 above)
Comment (MV88) [inline web/client/utils/MapInfoUtils.js]: No longer used in the rest of the file. So I removed as discussed.

File: web/client/utils/MapInfoUtils.js (hunk 4 above)
Comment (MV88) [inline web/client/utils/MapInfoUtils.js]: are you ok with this name getProjectedBBox ?

File: web/client/utils/MapInfoUtils.js (hunk 6 above)
Comment (MV88) [inline web/client/utils/MapInfoUtils.js]: what are you referring? if u mean reprojectBbox, is no longer used so it has been removed.

File: web/client/utils/MapInfoUtils.js (hunk 2 above)
Comment (MV88) [inline web/client/utils/MapInfoUtils.js]: createBBox?

File: web/client/utils/MapInfoUtils.js (hunk 6 above)
Comment (mbarto) [inline web/client/utils/MapInfoUtils.js]: I mean size

File: web/client/utils/MapInfoUtils.js (hunk 2 above)
Comment (mbarto) [inline web/client/utils/MapInfoUtils.js]: ok

File: web/client/utils/MapInfoUtils.js (hunk 4 above)
Comment (mbarto) [inline web/client/utils/MapInfoUtils.js]: yes

File: web/client/utils/MapInfoUtils.js (hunk 6 above)
Comment (MV88) [inline web/client/utils/MapInfoUtils.js]: it is used in getProjectedBBox or old (getForViewAndSize)

Comment (mbarto) [inline web/client/utils/MapInfoUtils.js]: It's ok, I didn't see it. Let's just not repeat 101 everywhere

File: web/client/utils/MapInfoUtils.js [hunk 7]
```diff
@@ -90,20 +77,87 @@ const MapInfoUtils = {
...
+     * Creates a new bbox.
+     * @param {number} minX Minimum X.
+     * @param {number} minY Minimum Y.
+     * @param {number} maxX Maximum X.
+     * @param {number} maxY Maximum Y.
+     * @param {Object} optExtent Destination extent.
+     * @return {Object Extent.
+     */
+    createBBox(minX, minY, maxX, maxY, optExtent) {
+        if (optExtent) {
```
Comment (mbarto) [inline web/client/utils/MapInfoUtils.js]: You can remove this block. The assign is enough

**Pull request merged**